qip-0015.md

```diff
@@ -0,0 +1,59 @@
+```
+  QIP: 15
+  Layer: Consensus (hard fork) 
+  Title: Qi UTXO Set Trimming
+  Author: jdowning100
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/quai-network/qips/wiki/Comments:QIP-00015
+  Status: Active
+  Type: Standards Track
+  Created: 2024-09-06
+  License: BSD-2-Clause
+```
+
+## Abstract
+
+The UTXO set is the set of all unspent transaction outputs in the Qi ledger. Qi is a money system with cash-like properties, such as denominations that can be broken down into smaller denominations. Combining smaller denominations into larger ones can be done with a different mechanism, but that is to be discussed in a separate QIP. This QIP proposes a mechanism to regularly trim dust UTXOs from the UTXO set.
+
+## Motivation
+
+The Qi UTXO set must be held on the disk of every node in the network in order for nodes to come into consensus, and the set can grow infinitely large. There are currently about [187.3 million UTXOs](https://www.blockchain.com/explorer/charts/utxo-count) in Bitcoin, with at ~70ish bytes per UTXO, takes up about 13gb of space on disk. 
+I fully believe that to build a monetary system for every individual the UTXO set for a given zone could grow to as large as 20 billion, which is about 200 UTXOs per user for 100 million users per zone. Qi UTXOs are about 57 bytes: 2 bytes for the leveldb prefix, 32 bytes for the txhash, 2 bytes for the index, 20 bytes for the address that may spend the UTXO, 1 byte for the denomination, and 1 byte or more for the lock (if applicable). This means that the UTXO set on disk could grow to as large as 1.14 TB. 
+Because the UTXO set can grow to be infinitely large, a well-funded attacker could launch a DDoS attack to grow the UTXO set to a size that is untenable for the average node to store on disk. In order to prevent this attack, this QIP proposes burning old, unused small-value UTXOs (below 1 Qi), such that the DDoS bloating attack would be expontentially more expensive (10 billion Qi vs 10 million Qi, for example). 
+
+## Specification
```

Comment:
Most of what you've written in this section focuses a lot on implementation details, but what the proposal really needs, is a description of the protocol. What are the new block acceptance criteria? What are the new transaction validity criteria? etc, etc.

This is not a spec for _how_ a particular client should implement the protocol.

**Move to the rationale section**
Some of this context is valuable as an explanation for _why_ you design the protocol a certain way, but most of this discussion should be removed. That stuff should be moved to the rationale section

**Still need a specification section**
The specification section should layout the new protocol rules, data types, etc. Some things this section might need to specify:
1) new block structure?
1) new block validity criteria?
1) new transaction structure?
1) new transaction validity criteria?

Reply:
1. new block structure? **None**
2. new block validity criteria? **As I explained, every block must trim UTXOs at a pre-defined rate**
3. new transaction structure? **None**
4. new transaction validity criteria? **None**